server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -392,8 +392,8 @@ private class ShardAllocationCounts {
         public void increment(ShardRouting routing, ClusterState state, NodesShutdownMetadata shutdowns, boolean verbose) {
             boolean isNew = isUnassignedDueToNewInitialization(routing);
             boolean isRestarting = isUnassignedDueToTimelyRestart(routing, shutdowns);
-            available &= routing.active() || isRestarting || isNew;
-            if ((routing.active() || isRestarting || isNew) == false) {
+            available &= routing.active() || isRestarting || isNew || routing.initializing();
```

Comment:
I think the original logic is correct here. An initializing primary shard should result in RED health. Under any normal circumstance initializing primaries should not be expected.

The problem in my mind is with the way "symptom" is calculated, it should (probably) also show initializing shards. I am not 100% up to date on what symptom mean compared to status though.

Reply:
Agreed, we reverted this back to what it was and we fixed the symptom and introduced a new diagnosis.